src/Umbraco.Core/Configuration/HealthChecks/ConfigurationService.cs

```diff
@@ -1,113 +1,65 @@
-                        Result = _textService.Localize("healthcheck/configurationServiceNodeNotFound", new[] { _xPath, _configFilePath })
-                    };
-
-                return new ConfigurationServiceResult
-                {
-                    Success = true,
-                    Result = string.Format(xmlNode.Value ?? xmlNode.InnerText)
-                };
-            }
-            catch (Exception ex)
-            {
-                _logger.Error<ConfigurationService>(ex, "Error trying to get configuration value");
-                return new ConfigurationServiceResult
-                {
-                    Success = false,
-                    Result = _textService.Localize("healthcheck/configurationServiceError", new[] { ex.Message })
-                };
-            }
+            _logger.Error<ConfigurationService>("A required ConfigurationService parameter was null", parameter);
+            throw new ArgumentNullException(parameter);
         }
 
         /// <summary>
-        /// Updates a value in a given configuration file with the given XPath
+        /// Updates a value in a given configuration file with the given path
         /// </summary>
         /// <param name="value"></param>
+        /// <param name="itemPath"></param>
         /// <returns></returns>
-        public ConfigurationServiceResult UpdateConfigFile(string value)
+        public ConfigurationServiceResult UpdateConfigFile(string value, string itemPath)
         {
             try
             {
-                if (File.Exists(_configFilePath) == false)
-                    return new ConfigurationServiceResult
-                    {
-                        Success = false,
-                        Result = _textService.Localize("healthcheck/configurationServiceFileNotFound", new[] { _configFilePath })
-                    };
-
-                var xmlDocument = new XmlDocument { PreserveWhitespace = true };
-                xmlDocument.Load(_configFilePath);
-
-                var node = xmlDocument.SelectSingleNode(_xPath);
-                if (node == null)
+                if (itemPath == null)
+                {
                     return new ConfigurationServiceResult
                     {
                         Success = false,
-                        Result = _textService.Localize("healthcheck/configurationServiceNodeNotFound", new[] { _xPath, _configFilePath })
+                        Result = _textService.Localize("healthcheck/configurationServiceNodeNotFound", new[] { itemPath, value })
                     };
+                }
 
-                if (node.NodeType == XmlNodeType.Element)
-                    node.InnerText = value;
-                else
-                    node.Value = value;
-
-                xmlDocument.Save(_configFilePath);
-                return new ConfigurationServiceResult { Success = true };
+                _configManipulator.SaveConfigValue(itemPath, value);
```

Comment:
I dont this this can be used directly. There are multiple possible issues..

1) This do not work if the value do not exists in the config. E.g using default value. This is also why we "merge" json objects when we insert connection string.
1) This only saves to the json file lowest in the config hierarchy (appsettings.json). But the value could be overridden in multiple other sources higher in the hierarchy (e.g. appsettings.Development.json).

Not sure how we can do, or maybe we should just remove the "Fix" option from some of the healthchecks?

Reply:
Good points. I think it is best if I don't try to fix in this PR, remove the fix option for the next push, and it can be considered again in a future product conversation - as discussed, whether we show a "How to fix" or actually try to write the fix.

Reply:
This is no longer being referenced, so:

- I've kept the code changes as if we were going to write the values
- I haven't deleted the service as it was pre-existing and created in 2016 for the first health checks